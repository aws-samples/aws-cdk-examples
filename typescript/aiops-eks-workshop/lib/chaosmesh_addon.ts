// lib/chaosmesh_addon.ts
import { Construct } from 'constructs';
import * as blueprints from '@aws-quickstart/eks-blueprints';
import { setPath } from '@aws-quickstart/eks-blueprints/dist/utils/object-utils';

/**
 * User provided options for the Helm Chart
 */
export interface ChaosMeshAddOnProps extends blueprints.HelmAddOnUserProps {
  version?: string,
  ingressEnabled?: boolean,
  chaosmeshServiceType?: string,
  chaosmeshDashboardListenPort?: string,
}

/**
 * Default props to be used when creating the Helm chart
 */
const defaultProps: blueprints.HelmAddOnProps & ChaosMeshAddOnProps = {
  name: "chaos-mesh",
  namespace: "chaos-testing",
  chart: "chaos-mesh",
  version: "2.4.1",
  release: "chaos-mesh",
  repository:  "https://charts.chaos-mesh.org",
  values: {},

  ingressEnabled: true,
  chaosmeshServiceType: "LoadBalancer",
  chaosmeshDashboardListenPort: "80"
};

/**
 * Main class to instantiate the Helm chart
 */
export class ChaosMeshAddOn extends blueprints.HelmAddOn {

  readonly options: ChaosMeshAddOnProps;

  constructor(props?: ChaosMeshAddOnProps) {
    super({...defaultProps, ...props});
    this.options = this.props as ChaosMeshAddOnProps;
  }

  deploy(clusterInfo: blueprints.ClusterInfo): Promise<Construct> {
    let values: blueprints.Values = populateValues(this.options);
    const chart = this.addHelmChart(clusterInfo, values);

    return Promise.resolve(chart);
  }
}

/**
 * populateValues populates the appropriate values used to customize the Helm chart
 * @param helmOptions User provided values to customize the chart
 */
function populateValues(helmOptions: ChaosMeshAddOnProps): blueprints.Values {
  const values = helmOptions.values ?? {};

  setPath(values, "ingress.enabled",  helmOptions.ingressEnabled);
  setPath(values, "dashboard.service.type",  helmOptions.chaosmeshServiceType);
  setPath(values, "dashboard.env.LISTEN_PORT", helmOptions.chaosmeshDashboardListenPort);
  setPath(values, "mysql.generate_passwords",  true);

  return values;
}
