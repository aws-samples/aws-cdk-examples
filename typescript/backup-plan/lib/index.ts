import {
  Duration,
  CfnOutput,
  aws_backup as bk,
  aws_events as events,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

export interface BackupProps {
  /**
   * Resources to apply backup plan.
   */
  readonly resources: bk.BackupResource[];

  /**
   * The display name of the backup plan.
   */
  readonly backupPlanName: string;

  /**
   * The duration after a backup job is successfully
   * started before it must be completed or it is
   * canceled by AWS Backup.
   *
   * Note: `backupCompletionWindow` must be at least 60 minutes greater
   * than @backupStartWindows
   *
   * @default - 3 hours
   */
  readonly backupCompletionWindow?: Duration;

  /**
   * The duration after a backup is scheduled before
   * a job is canceled if it doesn't start successfully.
   *
   * @default - 1 hour less than @backupCompletionWindow
   */
  readonly backupStartWindow?: Duration;

  /**
   * Specifies the duration after creation that a recovery point is deleted.
   * Must be greater than moveToColdStorageAfter.
   *
   * @default - 30 days
   */
  readonly deleteBackupAfter?: Duration;

  /**
   * Specifies the duration after creation that a recovery point is moved to cold storage.
   *
   * @default - recovery point is never moved to cold storage
   */
  readonly moveBackupToColdStorageAfter?: Duration;

  /**
   * How frequently backup jobs would be started.
   *
   * @default - 24 hours
   */
  readonly backupRateHour?: number;
}

/**
 * Construct to create a Backup Plan with specific backing cadence.
 *
 * @stability stable
 */
export class Backup extends Construct {
  /**
   * Backup plan
   */
  public readonly backupPlan: bk.BackupPlan;
  /**
   * 
   * @param scope Construct's scope
   * @param id Construct's id
   * @param props Construct's props
   */
  constructor(scope: Construct, id: string, props: BackupProps) {
    super(scope, id);

    const hourlyRate = `0/${props.backupRateHour || 24}`;

    const completionWindow = props.backupCompletionWindow || Duration.hours(3);
    const startWindow = props.backupStartWindow || Duration.hours(completionWindow.toHours() - 1);

    if ((completionWindow.toHours() - startWindow.toHours()) < 1)
      throw Error('Backup completion window must be at least 60 minutes greater than backup start window');

    const scheduledBkRule = new bk.BackupPlanRule({
      completionWindow,
      startWindow,
      deleteAfter: props.deleteBackupAfter || Duration.days(30),
      // Only cron expressions are supported
      scheduleExpression: events.Schedule.cron({
        minute: '0',
        hour: hourlyRate,
      }),
      moveToColdStorageAfter: props.moveBackupToColdStorageAfter,
    });

    this.backupPlan = new bk.BackupPlan(this, 'BackupPlan', {
      backupPlanName: props.backupPlanName,
      backupPlanRules: [scheduledBkRule],
    });

    this.backupPlan.addSelection('BackupSelection', {
      resources: props.resources,
    });

    // Outputs
    const outputVars = {
      BackupPlanId: this.backupPlan.backupPlanId,
      BackupPlanArn: this.backupPlan.backupPlanArn,
    };
    Object.entries(outputVars).forEach(
      ([outName, outValue]) => new CfnOutput(this, outName, { value: outValue }),
    );
  }
}
