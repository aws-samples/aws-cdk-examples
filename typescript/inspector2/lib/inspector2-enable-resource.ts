import * as cdk from 'aws-cdk-lib';
import {
  aws_iam as iam,
  custom_resources as custom_resources,
  aws_logs as logs,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

/** Properties for EnableInspector2Resource. */
export interface Inspector2EnableResourceProps extends cdk.ResourceProps {
  /**
   * Resource types to enable Amazon Inspector 2 scans for
   *
   * @default - 'ECR', 'EC2', 'LAMBDA'
   */
  readonly resourceTypes?: string[];
  /**
   * The number of days log events of the singleton Lambda function implementing
   * this custom resource are kept in CloudWatch Logs.
   *
   * @default logs.RetentionDays.INFINITE
   */
  readonly logRetention?: logs.RetentionDays;
}

/** A Custom Resource construct for enabling Amazon Inspector 2 in a single AWS account.
 * Please note that update operation is not supported. If you need to change the resource
 * types you need to recreate the resource (it's not a destructive operation).
 * 
 * For more details see https://docs.aws.amazon.com/inspector/latest/user/what-is-inspector.html
 */
export class Inspector2EnableResource extends cdk.Resource {
  constructor(scope: Construct, id: string, props: Inspector2EnableResourceProps) {
    super(scope, id);

    const inspector2Resources = props.resourceTypes ?? ['ECR', 'EC2', 'LAMBDA'];
    const physicalResourceId = custom_resources.PhysicalResourceId.fromResponse('accounts.0.accountId');

    const customResourceRole = new iam.Role(this, 'InspectorRole', {
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonInspector2FullAccess'),
      ],
    });

    new custom_resources.AwsCustomResource(this, 'Inspector2EnableResource', {
      resourceType: 'Custom::Inspector2Enable',
      role: customResourceRole,
      onCreate: {
        service: 'Inspector2',
        action: 'enable',
        parameters: {
          resourceTypes: inspector2Resources,
        },
        physicalResourceId,
      },
      // onUpdate: NOT supported
      onDelete: {
        service: 'Inspector2',
        action: 'disable',
        physicalResourceId,
        parameters: {
          resourceTypes: inspector2Resources,
        },
      },
      installLatestAwsSdk: true,
      logRetention: props.logRetention,
    });
  }
}
