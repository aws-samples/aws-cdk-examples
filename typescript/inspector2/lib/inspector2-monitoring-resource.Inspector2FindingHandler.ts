import {
  Finding
} from '@aws-sdk/client-inspector2';
import { EventBridgeEvent, EventBridgeHandler } from 'aws-lambda';

export const handler: EventBridgeHandler<'Inspector2 Finding', Finding, void> = async (event: EventBridgeEvent<'Inspector2 Finding', Finding>) => {
  console.log('Received Inspector2 Finding');
  const detail = event.detail;
  console.log(`[${detail.severity}] ${detail.awsAccountId} ${detail.title}  ${detail.type}`);
}
