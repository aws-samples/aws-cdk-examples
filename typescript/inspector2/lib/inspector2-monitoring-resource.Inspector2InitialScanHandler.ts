import * as AWSLambda from 'aws-lambda';

export interface InitialScan {
  'scan-status': string | undefined;
  'repository-name': string | undefined;
  'finding-severity-counts': {
    'CRITICAL': number;
    'HIGH': number;
    'MEDIUM': number;
  };
  'image-digest': string | undefined;
  'image-tags': Array<string> | undefined;
  'instance-id': string | undefined;
  'tags'?: Record<string, string>;
}

export async function handler(event: AWSLambda.EventBridgeEvent<'Inspector2 Scan', InitialScan>) {
  console.log('Received Inspector2 Initial Scan Results');
  console.log(event.detail);
}
