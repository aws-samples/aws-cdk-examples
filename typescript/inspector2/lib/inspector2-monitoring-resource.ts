import * as cdk from 'aws-cdk-lib';
import {
  aws_logs as logs,
  aws_events as events,
  aws_events_targets as targets,
  aws_lambda as lambda,
  aws_lambda_nodejs as lambda_nodejs,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

/** A Custom Resource construct for monitoring Amazon Inspector scan results.
 * 
 * Deploys a Lambda to handle initial scan results and another Lambda to handle Inspector2 Findings.
 * 
 * For more details see https://docs.aws.amazon.com/inspector/latest/user/findings-managing-automating-responses.html
 */
export class Inspector2MonitoringResource extends cdk.Resource {
  constructor(scope: Construct, id: string, props: cdk.ResourceProps) {
    super(scope, id, props);

    const initialScanRule = new events.Rule(this, 'initialScanRule', {
      eventPattern: {
        'detailType': ['Inspector2 Scan'],
        'source': ['aws.inspector2'],
        'detail': {
          'scan-status': ['INITIAL_SCAN_COMPLETE'],
        },
      },
    });

    const findingScanRule = new events.Rule(this, 'findingScanRule', {
      eventPattern: {
        'detailType': ['Inspector2 Finding'],
        'source': ['aws.inspector2'],
        'detail': {
          // Optionally filter by severity
          // 'severity': ['HIGH', 'CRITICAL'],
          'status': ['ACTIVE'],
        },
      },
    });

    const inspector2InitialScanHandler = new lambda_nodejs.NodejsFunction(scope, 'Inspector2InitialScanHandler', {
      timeout: cdk.Duration.minutes(15),
      runtime: lambda.Runtime.NODEJS_18_X,
      logRetention: logs.RetentionDays.ONE_DAY,
    });

    const inspector2FindingHandler = new lambda_nodejs.NodejsFunction(scope, 'Inspector2FindingHandler', {
      timeout: cdk.Duration.minutes(15),
      runtime: lambda.Runtime.NODEJS_18_X,
      logRetention: logs.RetentionDays.ONE_DAY,
    });

    initialScanRule.addTarget(new targets.LambdaFunction(inspector2InitialScanHandler));
    findingScanRule.addTarget(new targets.LambdaFunction(inspector2FindingHandler));
  }
}
