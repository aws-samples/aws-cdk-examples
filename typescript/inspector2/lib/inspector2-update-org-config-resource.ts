import * as cdk from 'aws-cdk-lib';
import {
  aws_iam as iam,
  custom_resources as custom_resources,
  aws_logs as logs,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

/** Properties for EnableInspector2Resource. */
export interface Inspector2UpdateOrganizationConfigurationResourceProps extends cdk.ResourceProps {
  /** Defines which scan types are enabled automatically for new members of your Amazon Inspector organization. */
  readonly autoEnable: {
    ec2: boolean;
    ecr: boolean;
    lambda: boolean;
  };

  /**
   * The number of days log events of the singleton Lambda function implementing
   * this custom resource are kept in CloudWatch Logs.
   *
   * @default logs.RetentionDays.INFINITE
   */
  readonly logRetention?: logs.RetentionDays;
}

/** A Custom Resource construct for updating organization configuration for Amazon Inspector.
 * 
 * The organization configuration can only be changed on the delegated admin account.
 * 
 * For more details see https://docs.aws.amazon.com/inspector/latest/user/adding-member-accounts.html
 */
export class Inspector2UpdateOrganizationConfigurationResource extends cdk.Resource {
  constructor(scope: Construct, id: string, props: Inspector2UpdateOrganizationConfigurationResourceProps) {
    super(scope, id);
    const physicalResourceId = custom_resources.PhysicalResourceId.of((Buffer.from(JSON.stringify(props)).toString('base64')));
    const autoEnableSdkCall: custom_resources.AwsSdkCall = {
      service: 'Inspector2',
      action: 'updateOrganizationConfiguration',
      parameters: {
        autoEnable: {
          ...props.autoEnable,
        },
      },
      physicalResourceId,
    };

    new custom_resources.AwsCustomResource(this, 'Inspector2UpdateOrganizationConfigurationResource', {
      resourceType: 'Custom::Inspector2OrganizationConfiguration',
      policy: custom_resources.AwsCustomResourcePolicy.fromSdkCalls({
        resources: custom_resources.AwsCustomResourcePolicy.ANY_RESOURCE,
      }),
      onCreate: autoEnableSdkCall,
      onUpdate: autoEnableSdkCall,
       // onDelete: Not needed,
      logRetention: props.logRetention,
      installLatestAwsSdk: true,
    });
  }
}
