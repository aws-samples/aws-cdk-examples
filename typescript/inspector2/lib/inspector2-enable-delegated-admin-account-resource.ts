import * as cdk from 'aws-cdk-lib';
import {
  aws_iam as iam,
  custom_resources as custom_resources,
  aws_logs as logs,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Inspector2EnableResource, Inspector2EnableResourceProps } from './inspector2-enable-resource';
import { Inspector2UpdateOrganizationConfigurationResource } from './inspector2-update-org-config-resource';

/** Properties for Inspector2EnableDelegatedAdminAccountResource. */
export interface Inspector2EnableDelegatedAdminAccountResourceProps extends cdk.ResourceProps {
  /**
   * The AWS account ID of the Amazon Inspector delegated administrator.
   */
  readonly delegatedAdminAccountId: string;

  /**
   * The number of days log events of the singleton Lambda function implementing
   * this custom resource are kept in CloudWatch Logs.
   *
   * @default logs.RetentionDays.INFINITE
   */
  readonly logRetention?: logs.RetentionDays;

  /** Properties for enabling Amazon Inspector 2 on the current account. */
  readonly inspector2EnableProps: Inspector2EnableResourceProps;

  /** Defines which scan types are enabled automatically for new members of your Amazon Inspector organization.
   * This is only used if the delegated admin account is the current account.
   */
  readonly autoEnable: {
    ec2: boolean;
    ecr: boolean;
    lambda: boolean;
  };
}

/**
 * A Custom Resource construct for enabling Amazon Inspector delegated administrator for your AWS Organizations organization.
 * 
 * The Custom Resource does three things:
 * - Enables Amazon Inspector
 * - Configures delegated administrator account
 * - If delegated account is current account: Updates organization configuration to auto-enable scan types
 * 
 * Note: Updating the resource (changing properties) is NOT supported due to how underlying APIs are implemented.
 * To change the properties, first delete the stack and then redeploy with new properties.
 * 
 * For more details see https://docs.aws.amazon.com/inspector/latest/user/designating-admin.html
 */
export class Inspector2EnableDelegatedAdminAccountResource extends cdk.Resource {
  constructor(scope: Construct, id: string, props: Inspector2EnableDelegatedAdminAccountResourceProps) {
    super(scope, id);

    const physicalResourceId = custom_resources.PhysicalResourceId.of((Buffer.from(JSON.stringify(props)).toString('base64')));

    // https://docs.aws.amazon.com/inspector/latest/user/designating-admin.html#delegated-admin-permissions
    const adminPolicy = new iam.PolicyDocument({
      statements: [new iam.PolicyStatement({
        sid: 'PermissionsForInspectorAdmin',
        effect: iam.Effect.ALLOW,
        actions: [
          'inspector2:EnableDelegatedAdminAccount',
          'inspector2:DisableDelegatedAdminAccount',
          'organizations:EnableAWSServiceAccess',
          'organizations:RegisterDelegatedAdministrator',
          'organizations:ListDelegatedAdministrators',
          'organizations:ListAWSServiceAccessForOrganization',
          'organizations:DescribeOrganizationalUnit',
          'organizations:DescribeAccount',
          'organizations:DescribeOrganization'
        ],
        resources: ['*']
      })]
    });
    const customResourceRole = new iam.Role(this, 'Inspector2EnableDelegatedAdminResourceRole', {
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
      inlinePolicies: {
        'PermissionsForInspectorAdmin': adminPolicy
      },
    });

    const enableInspector2 = new Inspector2EnableResource(this, 'Inspector2EnableResource', props.inspector2EnableProps);

    const enableDelegatedAdminAccountApiCall: custom_resources.AwsSdkCall = {
      service: 'Inspector2',
      action: 'enableDelegatedAdminAccount',
      parameters: {
        delegatedAdminAccountId: props.delegatedAdminAccountId,
      },
      physicalResourceId,
    };

    const delegatedAdminResource = new custom_resources.AwsCustomResource(this, 'Inspector2EnableDelegatedAdminAccountResource', {
      resourceType: 'Custom::Inspector2EnableDelegatedAdminAccount',
      role: customResourceRole,
      onCreate: enableDelegatedAdminAccountApiCall,
      onDelete: {
        service: 'Inspector2',
        action: 'disableDelegatedAdminAccount',
        physicalResourceId,
        parameters: {
          delegatedAdminAccountId: props.delegatedAdminAccountId,
        },
      },
      installLatestAwsSdk: true,
      logRetention: props.logRetention,
    });
    delegatedAdminResource.node.addDependency(enableInspector2);

    if (props.delegatedAdminAccountId === cdk.Stack.of(this).account) {
      new Inspector2UpdateOrganizationConfigurationResource(this, 'Inspector2UpdateOrganizationConfigurationResource', {
        logRetention: props.logRetention,
        autoEnable: {
          ...props.autoEnable,
        }
      }).node.addDependency(delegatedAdminResource);
    }
  }
}
