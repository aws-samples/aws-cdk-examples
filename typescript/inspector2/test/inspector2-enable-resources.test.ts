import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { Inspector2EnableResource } from '../lib/inspector2-enable-resource';

test('Inspector2EnableResource creates required resources', () => {
  const app = new cdk.App();

  const stack = new cdk.Stack(app, 'Inspector2EnableStack', { });
  new Inspector2EnableResource(stack, 'EnableInspector2Resource', {
    resourceTypes: ['ECR', 'EC2', 'LAMBDA'],
    logRetention: cdk.aws_logs.RetentionDays.ONE_DAY,
  });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});
