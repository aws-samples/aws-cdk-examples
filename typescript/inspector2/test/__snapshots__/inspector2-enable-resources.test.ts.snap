// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inspector2EnableResource creates required resources 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": {
      "DependsOn": [
        "EnableInspector2ResourceInspectorRole75753456",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "6bbce4c77566d179835e0a6de3293a27d9021f2f49d0055bb8e465eb6e1d89b1.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "EnableInspector2ResourceInspectorRole75753456",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287LogRetentionCE72797A": {
      "Properties": {
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
        "ServiceToken": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "EnableInspector2ResourceInspector2EnableResource58B4E7AE": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Create": "{"service":"Inspector2","action":"enable","parameters":{"resourceTypes":["ECR","EC2","LAMBDA"]},"physicalResourceId":{"responsePath":"accounts.0.accountId"}}",
        "Delete": "{"service":"Inspector2","action":"disable","physicalResourceId":{"responsePath":"accounts.0.accountId"},"parameters":{"resourceTypes":["ECR","EC2","LAMBDA"]}}",
        "InstallLatestAwsSdk": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
      },
      "Type": "Custom::Inspector2Enable",
      "UpdateReplacePolicy": "Delete",
    },
    "EnableInspector2ResourceInspectorRole75753456": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonInspector2FullAccess",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": {
      "DependsOn": [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "eb5b005c858404ea0c8f68098ed5dcdf5340e02461f149751d10f59c210d5ef8.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": [
          {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
